homeassistant/components/recorder/table_managers/states_meta.py

```diff
@@ -24,7 +24,7 @@
 class StatesMetaManager(BaseLRUTableManager[StatesMeta]):
     """Manage the StatesMeta table."""
 
-    active = False
+    active = True
```

Comment:
Set to True because the data migration happens before recorder starts.
In a follow-up PR which removes the legacy history API queries, this flag will be removed.